frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,630 @@
+        outlinecolor: colors.transparent,
+        outlinewidth: 8,
+        len: 0.75,
+        y: 0.5745,
+        title: {
+          font: {
+            color: getGray70(theme),
+            size: fontSizes.smPx,
+          },
+        },
+        tickfont: {
+          color: getGray70(theme),
+          size: fontSizes.twoSmPx,
+        },
+      },
+      colorscale: {
+        ...(hasLightBackgroundColor(theme)
+          ? {
+              diverging: divergingColorscaleLightTheme,
+              sequential: sequentialColorscaleLightTheme,
+              // reverse to dark for sequential minus
+              sequentialminus: sequentialColorscaleDarkTheme,
+            }
+          : {
+              diverging: divergingColorscaleDarkTheme,
+              sequential: sequentialColorscaleDarkTheme,
+              // reverse to light for sequential minus
+              sequentialminus: sequentialColorscaleLightTheme,
+            }),
+      },
+    },
+    colorscale: {
+      ...(hasLightBackgroundColor(theme)
+        ? {
+            diverging: divergingColorscaleLightTheme,
+            sequential: sequentialColorscaleLightTheme,
+            // reverse to dark for sequential minus
+            sequentialminus: sequentialColorscaleDarkTheme,
+          }
+        : {
+            diverging: divergingColorscaleDarkTheme,
+            sequential: sequentialColorscaleDarkTheme,
+            // reverse to light for sequential minus
+            sequentialminus: sequentialColorscaleLightTheme,
+          }),
+    },
+    // specifically for the ternary graph
+    ternary: {
+      gridcolor: getGray70(theme),
+      bgcolor: colors.bgColor,
+      title: {
+        font: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.smPx,
+        },
+      },
+      color: getGray70(theme),
+      aaxis: {
+        gridcolor: getGray70(theme),
+        linecolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+      baxis: {
+        linecolor: getGray70(theme),
+        gridcolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+      caxis: {
+        linecolor: getGray70(theme),
+        gridcolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+    },
+  }
+
+  // cant use merge. use assign because plotly already has properties defined.
+  assign(layout, streamlitTheme)
+}
+
+/**
+ * Applies the colorscale, colors for unique graphs, and discrete coloring
+ * for in general through assigning properties in spec.data
+ * @param data - spec.data
+ */
+export function applyStreamlitThemeData(data: any, theme: Theme): void {
+  applyColorscale(data, theme)
+  applyUniqueGraphColorsData(data, theme)
+  changeDiscreteColors(data, theme)
```

Comment:
nit: is there a reason via this one is called `change...` and the other methods all use the `apply` prefix?